docs/ARCHITECTURE.MD

```diff
@@ -1,72 +1,287 @@
+
+Taking the code from a Git repository and building it on an OpenShift cluster takes a lot of time 
+mainly because all dependencies have to be downloaded from the Maven public repository.
+
+To overcome this problem, the recommendation is to have a Maven cache available such as a Nexus 
+server. The problem with this approach is that developers should set up another infrastructure 
+component to speed up the build process, which could be cumbersome and resource hungry.
+
+Another approach is something that we are working on ([KOGITO-602](https://issues.redhat.com/browse/KOGITO-602)),
+which is to provide the Kogito dependencies libraries that Red Hat already distribute within the 
+s2i images, like Quarkus. This won’t require a Nexus server ready and will speed up the build 
+process by nearly 70%. The problem with this solution is that the build images will be 
+considerably large and Spring Boot based services won’t gain much since we can’t redistribute 
+their libraries with our images.
+
+Since native builds are also supported by Kogito Services based on Quarkus, when building such 
+services on OpenShift, it will require a sizable amount of memory and CPU resources. 
+In local development scenarios this could be nearly impossible to achieve, since in our tests a 
+small [example can take up to 10 GB of memory to build](https://github.com/kiegroup/kogito-cloud-operator/#native-x-jvm-builds). 
+
+#### Local Builds
+
+A more affordable solution in development scenarios is for the developer to provide their already
+built artifacts. To support such scenarios, we created the Binary Build solution, which is a 
+specially configured image and OpenShift’s build configuration that takes a local built artifact
+and uploads it to the cluster. For more information, please refer see the [Operator Documentation](https://github.com/kiegroup/kogito-cloud-operator/#binary-builds). 
+
+Since we are aiming for Kubernetes support, Binary Build is not a supported solution since 
+`BuildConfig` is an OpenShift exclusive object. In such scenarios, not only the Kogito service 
+binary needs to be provided by the developer/administrator, but the entire image.
+
+One approach is to have a template Dockerfile being generated for the developer once they create 
+a Kogito project via the Maven plugin. This Dockerfile will be based on the Kogito UBI image with 
+a new layer to copy the generated project’s artifact and library from the target dir.
+
+Another way of achieving local builds is to use the s2i client instead of a Dockerfile. 
+Since building and maintaining Dockerfiles is a well-known approach, it’s preferable over the 
+s2i client.
+
+Since we have the “Kogito CLI” tool that facilitates the integration with the operator, 
+one possible solution is to incorporate the local image build to it. This way, the full circle 
+could be achieved by the tool:
+
+1. Kogito CLI Generates the Kogito project’s skeleton
```

Comment:
Is this going to be provided by Kogito CLI too?
Anyway user can use also Maven archetype to generate the skeleton.

Reply:
This is just a possibility/we speculating about the future. :)